### PR TITLE
refactor(query): If table_id is specified, use it directly

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Note:

If len of table_ids gt 10, also use list_table

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/16098)

### CODE REVIEW

File: src/query/storages/system/src/tables_table.rs [hunk 1]
```diff
@@ -245,6 +246,18 @@ where TablesTable<T, U>: HistoryAware
                                     db_name.push(database.clone());
                                 }
                             }
+                        } else if col_name == "table_id" {
+                            if let Scalar::String(tid) = scalar {
```
Comment (sundy-li) [inline src/query/storages/system/src/tables_table.rs:250]: table_id could be int/string

File: src/query/storages/system/src/tables_table.rs [hunk 2]
```diff
@@ -245,6 +248,27 @@ where TablesTable<T, U>: HistoryAware
...
```
Comment (sundy-li) [inline src/query/storages/system/src/tables_table.rs:252]: use `check_number` to get the tid.

Comment (TCeason) [reply src/query/storages/system/src/tables_table.rs:252]: How about as_u_int64()? NumberScalar support EnumAsInner

Comment (TCeason) [reply src/query/storages/system/src/tables_table.rs:252]: Done

File: tests/sqllogictests/suites/ee/03_ee_vacuum/03_0000_vacuum_ctas.test [hunk 3]
```diff
@@ -16,7 +16,7 @@ statement error 1006
 create or replace table t (c int) 'fs:///tmp/ctas/' as select number / (number-3999999) from numbers(4000000);
 
 # there should be no table ctas_test.t
-query I
+query T
```
Comment (sundy-li) [inline tests/sqllogictests/suites/ee/03_ee_vacuum/03_0000_vacuum_ctas.test:19]: I think this file should not be changed.

**Review by sundy-li — APPROVED** [review]

File: src/query/storages/system/src/tables_table.rs [hunk 4]
```diff
@@ -245,7 +250,28 @@ where TablesTable<T, U>: HistoryAware
...
+                                &FunctionContext::default(),
+                                &Expr::<usize>::Constant {
+                                    span: None,
+                                    scalar: scalar.clone(),
+                                    data_type: scalar.as_ref().infer_data_type(),
+                                },
+                                &BUILTIN_FUNCTIONS,
+                            ) {
+                                Ok(id) => tables_ids.push(id),
+                                Err(_) => invalid_tables_ids = true,
```
Comment (b41sh) [inline src/query/storages/system/src/tables_table.rs:265]: If the table id is not u64, the query conditions are not met. Can we return an empty Block?

Comment (TCeason) [reply src/query/storages/system/src/tables_table.rs]: No. maybe this : name = 'xx' or table_id = 123.1.

Comment (b41sh) [reply src/query/storages/system/src/tables_table.rs]: ok

**Review by b41sh — APPROVED** [review]

**Comment by TCeason** [comment]
cc @BohuTANG Please review this pr.

cc @wubx wait this pr merge, you can try it.

**Pull request merged**